ethcore/src/client/client.rs

```diff
@@ -837,6 +837,23 @@ impl snapshot::DatabaseRestore for Client {
 	}
 }
 
+/// Find transition point between `lower` and `upper` where `cond` changes from `false` to `true`.
+/// Returns the lowest value between `lower` and `upper` for which `cond` returns true.
+/// We assert: `cond(lower) = false`, `cond(upper) = true`
+pub fn binary_chop<F>(mut lower: U256, mut upper: U256, mut cond: F) -> U256 where F: FnMut(U256) -> bool {
```

Comment:
Does this really belong in `client`? Should either be in a utility file or an inline function within `estimate_gas` if that's truly the only place it's useful.